pandas/io/tests/test_common.py

```diff
@@ -139,3 +139,10 @@ def test_next(self):
             self.assertEqual(next_line.strip(), line.strip())
 
         self.assertRaises(StopIteration, next, wrapper)
+
+    def test_cparser_error_depr(self):
+        # see gh-12665
+        with tm.assert_produces_warning(DeprecationWarning,
+                                        check_stacklevel=False):
+            with tm.assertRaises(common.ParserError):
+                raise common.CParserError()
```

Comment:
And some usages I found seem to also import this from `pandas.parser`, so maybe we should keep it there as well (so not only in `io.common`).

https://github.com/glue-viz/glue/blob/e9742059112d00e562fc3123703678d66fad1996/glue/core/data_factories/pandas.py#L60 
https://github.com/sawadyrr5/pyjsf/blob/cf1e8643afb65912237483eac5a2735df5a8a8a6/pyjsf/data/download.py